### PR TITLE
Add shared functions for modules/subworkflows install command

### PR DESCRIPTION
**Pull request by mirpedrol**
Follow up from #1916 
Add shared helper functions for `nf-core modules install` and `nf-core subworkflows install` commands.

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/components/components_install.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import rich
+
+import nf_core.modules.modules_utils
+import nf_core.utils
+from nf_core.modules.modules_repo import NF_CORE_MODULES_NAME
+
+log = logging.getLogger(__name__)
+
+
+def verify_sha(modules_repo, prompt, sha):
```
Comment (awgymer) [inline nf_core/components/components_install.py:18]: I think this should be moved to a method on `ModulesRepo`. It doesn't have any specific component utility and always takes an instance of one as the first arg anyway.

File: nf_core/components/components_install.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+def get_version(component, component_type, sha, prompt, current_version, modules_repo):
+    """
+    Get the version to install
+    """
+    if sha:
+        version = sha
+    elif prompt:
+        try:
+            version = nf_core.modules.modules_utils.prompt_module_version_sha(
```
Comment (mashehu) [inline nf_core/components/components_install.py:100]: should we move `prompt_module_version_sha` to component_utils and make it `prompt_component_version_sha`?

File: nf_core/components/components_install.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            return False
+    else:
+        # Fetch the latest commit for the module
+        version = modules_repo.get_latest_component_version(component, component_type)
+    return version
+
+
+def clean_modules_json(component, component_type, modules_repo, modules_json):
+    """
+    Remove installed version of module/subworkflow
```
Comment (mashehu) [inline nf_core/components/components_install.py:116]:
```suggestion
    Remove installed version of module/subworkflow from modules.json
```

File: nf_core/modules/update.py [hunk 4]
```diff
@@ -155,7 +155,7 @@ def update(self, module=None):
                     module, modules_repo=modules_repo, installed_sha=current_version
                 )
             else:
-                version = modules_repo.get_latest_module_version(module)
+                version = modules_repo.get_latest_component_version(module, "modules")
```
Comment (mashehu) [inline nf_core/modules/update.py:158]:
```suggestion
                version = modules_repo.get_latest_component_version(module, self.component_type)
```

File: nf_core/modules/modules_utils.py [hunk 5]
```diff
@@ -201,7 +201,7 @@ def prompt_module_version_sha(module, modules_repo, installed_sha=None):
     git_sha = ""
     page_nbr = 1
 
-    all_commits = modules_repo.get_module_git_log(module)
+    all_commits = modules_repo.get_component_git_log(module, "modules")
```
Comment (mashehu) [inline nf_core/modules/modules_utils.py:204]:
```suggestion
    all_commits = modules_repo.get_component_git_log(module,self.component_type)
```

**Review by mashehu — APPROVED** [review]

File: nf_core/modules/modules_utils.py (hunk 5 above)
Comment (awgymer) [reply nf_core/modules/modules_utils.py:204]: This doesn't have access to `self` or `component_type` - it's a helper function in`module_utils`. Unless you are meaning to do this only if we move the method?

Comment (mashehu) [reply nf_core/modules/modules_utils.py:204]: ah, sorry, looked like it was in the same context as the one above. 👍🏻

File: nf_core/components/components_install.py (hunk 2 above)
Comment (mashehu) [reply nf_core/components/components_install.py]: done in [23c457c](https://github.com/nf-core/tools/pull/1980/commits/23c457ce80465911bb28ad44568574ba9c6cf726)

**Pull request merged**